src/main.rs

```diff
@@ -144,3 +154,66 @@ fn aishrc_path() -> Result<PathBuf, std::io::Error> {
     let aishrc_path: PathBuf = home.join(".aishrc");
     Ok(aishrc_path)
 }
+
+fn expand_aliases(aliases: &HashMap<String, String>, buffer: &str) -> String {
+    let mut temp_aliases = aliases.clone();
+
+    buffer
+        .split(';')
+        .map(|command| {
+            command
+                .split("&&")
+                .map(|and_command| {
+                    and_command
+                        .split('|')
+                        .map(|pipe_command| expand_single_command(&mut temp_aliases, pipe_command))
+                        .collect::<Vec<_>>()
+                        .join(" | ")
+                })
+                .collect::<Vec<_>>()
+                .join(" && ")
+        })
+        .collect::<Vec<_>>()
+        .join("; ")
+}
+
+fn expand_single_command(temp_aliases: &mut HashMap<String, String>, command: &str) -> String {
+    let mut words: Vec<String> = command.split_whitespace().map(String::from).collect();
+
+    if words.is_empty() {
+        return String::new();
+    }
+
+    if words[0] == "alias" && words.len() > 1 {
+        handle_alias_definition(temp_aliases, &words[1..]);
+        return command.trim().to_string();
+    }
+
+    expand_command_aliases(temp_aliases, &mut words);
+    words.join(" ")
+}
+
+fn handle_alias_definition(temp_aliases: &mut HashMap<String, String>, args: &[String]) {
+    let full_command = args.join(" ");
+    if let Some(equals_pos) = full_command.find('=') {
+        let (alias_name, alias_value) = full_command.split_at(equals_pos);
+        temp_aliases.insert(
+            alias_name.to_string(),
+            alias_value[1..].trim_matches('\'').to_string(),
+        );
+    }
+}
```

Comment:
What is this for?

Reply:
It's solving the problem where we have something like `alias greet='echo \"Hello, World!\"' && greet` because expand tokens gets called before we add greet to the aliases, the second greet after the && doesn't get expanded so this checks if we have an alias on the left of the command and if we do then it expands the right hand side. So basically all handle_alias_definition is doing is getting the alias greet and the expansion of this which is echo \"Hello, World!\", this is then returned and used in expand_command_aliases to turn the whole command into `alias greet='echo \"Hello, World!\"' && alias greet='echo \"Hello, World!\"`